common/OpTestUtil.py

```diff
@@ -1913,6 +1913,112 @@ def wait_for(self, func, timeout, first=0.0, step=1.0, text=None, args=None, kwa
+        except CommandFailed as cmd_failed:
+            if cmd_failed.exitcode == 1 and len(cmd_failed.output) == 0:
+                pass
+            else:
+                raise cmd_failed
+        dmsg_log = self.skip_dmesg_messages(out, skip_errors) if skip_errors else out
+        if dmsg_log:
+            if output_file:
+                output_dir = os.path.join(host.results_dir, "DmesgErrorLogs")
+                if not os.path.exists(output_dir):
+                    os.makedirs(output_dir)
+                output_file = os.path.join(output_dir, output_file)
+                with open(output_file, "w+", encoding='utf-8') as log_f:
+                    log_f.write(dmsg_log)
+                err = "Found failures in dmesg. Please check dmesg log %s." % (output_file)
+            else:
+                err = "Found failures in dmesg. Please check debug log."
+                log.debug(dmsg_log)
+            self.gather_os_logs(output_dirname="TestFailLogs")
+            raise OpTestError("Test failed. {}".format(err))
+
+    def skip_dmesg_messages(self, dmesg, skip_messages):
+        """
+        Remove some messages from a dmesg buffer.
+
+        This method will remove some lines in a dmesg buffer if some strings are
+        present. Returning the same buffer, but with less lines (in case of match).
+
+        :dmesg: dmesg messages from which filter should be applied. This
+                must be a decoded output buffer with new lines.
+        :type dmesg: str
+        :skip_messages: list of strings to be removed
+        :type skip_messages: list
+        """
+        def filter_strings(line):
+            return not any([string in line for string in skip_messages])
+
+        return '\n'.join(filter(None, filter(filter_strings, dmesg.splitlines())))
+
+    def gather_os_logs(self, list_of_files=[], list_of_commands=[], output_dirname=None):
+        host = self.conf.host()
+        if not output_dirname:
+            output_dirname = "Logs_%s" % (time.asctime(time.localtime())).replace(" ", "_")
+        output_dir = os.path.join(host.results_dir, output_dirname)
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+        self.__gather_files_log(list_of_files, output_dir)
+        self.__gather_command_log(list_of_commands, output_dir)
+
+    def __gather_files_log(self, list_of_files=[], output_dir=None):
+        host = self.conf.host()
+        default_files = ['/var/log/messages', '/var/log/boot.log']
+        list_of_files.extend(default_files)
+
+        try:
+            host.host_run_command("mkdir -p fileLogs")
+            for file in set(list_of_files):
+                fn = "%s.log" % '-'.join(file.strip(os.path.sep).split(os.path.sep))
+                host.host_run_command("cp %s fileLogs/%s" % (file, fn))
+            host.copy_files_from_host(sourcepath=output_dir, destpath="fileLogs")   
+            host.host_run_command("rm -rf fileLogs")
```

Comment:
rename fileLogs to just `sysinfo` and inside the  /sysinfo/files/    /sysinfo/commands/